### ISSUE TITLE
make errors after upgrade to Docker Compose 2.6.0

### ISSUE THREAD

**Issue by divbzero** [pypa/warehouse#11674]
Steps to reproduce:
1. Upgrade to Docker Compose 2.6.0
2. Run `make serve` for pypa/warehouse.

Error message:
```sh
Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "hupper": executable file not found in $PATH: unknown
make: *** [serve] Error 1
```

**Comment by divbzero** [comment]
When I ran `docker-compose run worker bash` and `echo $PATH` I saw my host machine’s `$PATH` echoed in the container. This appears to be an issue with Docker Compose 2.6.0 mishandling environment variables from `env_file`:
- docker/compose#9521

I’m opening and closing this issue as here in case other Warehouse devs encounter the same issue.

**Comment by divbzero** [comment]
Reverting to [Docker Desktop 4.8.2](https://docs.docker.com/desktop/release-notes/#docker-desktop-482) resolved the issue.

**Comment by divbzero** [comment]
This environment variable issue was fixed by docker/compose#9636 (commit [6d613c8](https://github.com/docker/compose/commit/6d613c8)) which was released with [Docker Compose 2.8.0](https://github.com/docker/compose/releases/tag/v2.8.0) and [Docker Desktop 4.12.0](https://docs.docker.com/desktop/release-notes/#docker-desktop-4120).
